dataproxy/service.go

```diff
@@ -87,6 +91,54 @@ func (s Service) CreateUploadLocation(ctx context.Context, req *service.CreateUp
 	}, nil
 }
 
+// CreateDownloadLink retrieves the requested artifact type for a given execution (wf, node, task) as a signed url(s).
+func (s Service) CreateDownloadLink(ctx context.Context, req *service.CreateDownloadLinkRequest) (
+	resp *service.CreateDownloadLinkResponse, err error) {
+	if req, err = s.validateCreateDownloadLinkRequest(req); err != nil {
+		return nil, err
+	}
+
+	// Lookup task, node, workflow execution
+	var nativeURL string
+	switch o := req.GetSource().(type) {
```

Comment:
This also seems to be validation against the existence of node execution for the request.
Should we move this also to validateCreateDownloadLinkRequest. May be specific step in that which caters to node  validation so this function just has validation and creation of signedUrl

Reply:
There is already a check for the existence of Source in the validation method...